modules/scripts/startup-script/main.tf

```diff
@@ -89,9 +88,9 @@ resource "google_storage_bucket" "configs_bucket" {
 resource "google_storage_bucket_object" "scripts" {
   # this writes all scripts exactly once into GCS
   for_each = local.runners_map
-  name     = "${local.storage_folder_path_prefix}${each.key}"
-  content  = each.value["content"]
-  source   = each.value["source"]
+  name     = "${local.storage_folder_path_prefix}${each.key}-${try(md5(each.value.content), filemd5(each.value.source))}"
```

Comment:
I think it will be more user friendly if we truncate. I am thinking terraform cli output, pantheon ui, etc...

md5 spits out 32 chars. I think 6 or 7 should be more than sufficient (for ref git short sha is 7).

Reply:
I understand that the git short sha is calculated dynamically based on the repo size.  But 7 is a reasonable guidepost.  I personally think there is value in knowing the md5sum from the object name and the line becomes a bit messy if you add substr inside of it.

But if you feel strongly about it, let's pick the first 8 characters as it divides evenly into 32, and do it once correctly. WDYT?

Reply:
If you think others would get value out of having the full md5 then you can add variable that lets you configure this functionality. But I don't think having a longer hash provides any value.

If the line gets too long and cannot be effectively wrapped, you can calculate it in a local block and add it to a dict or create a new dict.

I don't see why dividing evenly is necessary for truncating. I would argue that even 7 is far more than needed since we are really only ever distinguishing between two versions at a time. You could go with 4 digits. Any of these options would be better than 32.